server/src/Application/Services/Flashcard/FlashcardFileImporter.cs

```diff
@@ -0,0 +1,67 @@
+using System.Collections.Concurrent;
+using StudyZen.Application.Dtos;
+
+namespace StudyZen.Application.Services
+{
+    public class FlashcardFileImporter
+    {
+        private readonly IFlashcardService _flashcardService;
+
+        public FlashcardFileImporter(IFlashcardService flashcardService)
+        {
+            _flashcardService = flashcardService;
+        }
+
+        public void ImportFlashcardsFromCsvStream(Stream stream, int flashcardSetId)
+        {
+            try
+            {
+                var lines = ReadCsvLinesFromStream(stream);
+
+                var flashcardsToCreate = new BlockingCollection<CreateFlashcardDto>();
+
+                Parallel.ForEach(lines, line =>
+                {
+                    var values = line.Split(',');
+
+                    if (values.Length == 2)
```

Comment:
I think it would be valid to throw an exception since the input is incorrect. Throwing a custom exception would help to map it to an error, knowing that the HTTP status code should be 4xx and not 5xx.